common/src/main/java/com/wynntils/services/mapdata/providers/builtin/PlayerProvider.java

```diff
@@ -0,0 +1,266 @@
+        reloadHadesUsers();
+    }
+
+    @SubscribeEvent
+    public void onHadesDisconnectedEvent(HadesEvent.Disconnected event) {
+        reloadHadesUsers();
+    }
+
+    @SubscribeEvent
+    public void onHadesUserAddedEvent(HadesUserEvent.Added event) {
+        reloadHadesUsers();
+    }
+
+    @SubscribeEvent
+    public void onHadesUserRemovedEvent(HadesUserEvent.Removed event) {
+        reloadHadesUsers();
+    }
+
+    private void reloadHadesUsers() {
+        PLAYERS.forEach(this::notifyCallbacks);
+        PLAYERS.clear();
+        Services.Hades.getHadesUsers().map(RemotePlayerLocation::new).forEach(PLAYERS::add);
+    }
+
+    @Override
+    public String getProviderId() {
+        return "players";
+    }
+
+    @Override
+    public Stream<MapFeature> getFeatures() {
+        return PLAYERS.stream().map(MapFeature.class::cast);
+    }
+
+    @Override
+    public Stream<MapCategory> getCategories() {
+        return PROVIDED_CATEGORIES.stream();
+    }
+
+    private static final class PlayersCategory implements MapCategory {
+        @Override
+        public String getCategoryId() {
+            return "wynntils:player";
+        }
+
+        @Override
+        public Optional<String> getName() {
+            return Optional.of("Player positions");
+        }
+
+        @Override
+        public Optional<MapAttributes> getAttributes() {
+            return Optional.of(new AbstractMapAttributes() {
+                @Override
+                public Optional<Integer> getPriority() {
+                    return Optional.of(900);
+                }
+
+                @Override
+                public Optional<MapVisibility> getIconVisibility() {
+                    return Optional.of(FixedMapVisibility.ICON_ALWAYS);
+                }
+            });
+        }
+    }
+
+    public static final class RemotePlayerLocation implements MapLocation {
+        private final HadesUser hadesUser;
+
+        public RemotePlayerLocation(HadesUser hadesUser) {
+            this.hadesUser = hadesUser;
+        }
+
+        @Override
+        public Location getLocation() {
+            return new Location((int) hadesUser.getX(), (int) hadesUser.getY(), (int) hadesUser.getZ());
+        }
+
+        @Override
+        public String getFeatureId() {
+            return hadesUser.getName().toLowerCase(Locale.ROOT);
+        }
+
+        @Override
+        public String getCategoryId() {
+            return "wynntils:players:" + (hadesUser.getRelation().isEmpty() ? "other" : hadesUser.getRelation());
+        }
+
+        @Override
+        public Optional<MapAttributes> getAttributes() {
+            return Optional.of(new AbstractMapAttributes() {
+                @Override
+                public Optional<String> getLabel() {
+                    return Optional.of(hadesUser.getName());
+                }
+
+                @Override
+                public Optional<CustomColor> getLabelColor() {
```

Comment:
This can move to the category. In general, a feature should override/define *only* the attributes that are essential to have different for this particular feature. In this case, overriding label is correct. (And in theory icon, if we could have fixed so that we could actually provide an icon.)

Basically, we need to add a category `wynntils:player:friend` which sets label color to the relation color of friend, etc. 

I think this means that the knowledge of coding hades relation to color can reside purely in the category (as it should), and that hadesUser.getRelationColor() can be removed.